### PR TITLE
Move all the extensions interface under a same directory

### LINKED ISSUE #599: [FEATURE] Move all the extensions interface under a same directory
### Is your feature request related to a problem?
Current, the extension interfaces are under `org.opensearch.src` directly, we can create a directory under `src`, call it as `extensions` and move these files under it. Also, we have to take care of moving the tests with the same pattern

1. [PersistentTaskExtension](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/ScriptExtension.java)
2. [ReloadableExtension](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/ReloadableExtension.java)
3. [IndexStoreExtension](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/IndexStoreExtension.java)
4. [ScriptExtension](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/ScriptExtension.java)
5. [IngestExtension](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/IngestExtension.java)
6. [IndexStoreExtension](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/IndexStoreExtension.java)
7. [SearchExtension](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/SearchExtension.java)
8. [SystemIndexExtension](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/SystemIndexExtension.java)
9. [MapperExtension](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/MapperExtension.java)
10. [RepositoryExtension](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/RepositoryExtension.java)

### What solution would you like?
_A clear and concise description of what you want to happen._

### What alternatives have you considered?
_A clear and concise description of any alternative solutions or features you've considered._

### Do you have any additional context?
_Add any other context or screenshots about the feature request here._

### PR DESCRIPTION
**Pull request by kokibas**
### Description
Move all the extensions interface under a same directory

### Issues Resolved
#599 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/sdk/ActionExtension.java [hunk 1]
```diff
@@ -1,191 +0,0 @@
-/*
- * Copyright OpenSearch Contributors
- * SPDX-License-Identifier: Apache-2.0
- *
- * The OpenSearch Contributors require contributions made to
- * this file be licensed under the Apache-2.0 license or a
- * compatible open source license.
- */
-
-package org.opensearch.sdk;
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/sdk/ActionExtension.java:10]: This should have been moved into `org.opensearch.sdk.api` with all the rest.  Where did it go?

File: CREATE_YOUR_FIRST_EXTENSION.md [hunk 2]
```diff
@@ -106,7 +106,7 @@ But it doesn't _do_ anything yet.  Here is where you can start defining your own
 If you want to handle REST Requests, implement the `ActionExtension` interface and override the `getExtensionRestHandlers()` method.  Pass a list of classes which will handle those requests.
 
 ```java
-import org.opensearch.sdk.ActionExtension;
+import org.opensearch.sdk.RepositoryExtension.ActionExtension;
```
Comment (dbwiddis) [inline CREATE_YOUR_FIRST_EXTENSION.md:109]: I"m not sure what `RepositoryExtension` is as it's not part of this PR.  But that class should have been moved into the `.api` package with all the rest, and all these import changes should be `import org.opensearch.sdk.api.ActionExtension;`

File: src/test/java/org/opensearch/sdk/TestExtensionInterfaces.java [hunk 3]
```diff
@@ -190,23 +190,4 @@ public void testRepositoryExtension() {
         assertTrue(repositoryExtension.getRepositories(null, null, null, null).isEmpty());
         assertTrue(repositoryExtension.getInternalRepositories(null, null, null, null).isEmpty());
     }
-
-    @Test
```
Comment (dbwiddis) [inline src/test/java/org/opensearch/sdk/TestExtensionInterfaces.java:194]: we shouldn't be deleting tests

**Review by dbwiddis — CHANGES_REQUESTED** [review]
I'm not sure what the `RepositoryExtension` class is or why ActionExtension is there.  It should be just like all the other extensions and get moved into the `.api` directory.

File: CREATE_YOUR_FIRST_EXTENSION.md [hunk 4]
```diff
@@ -106,7 +106,7 @@ But it doesn't _do_ anything yet.  Here is where you can start defining your own
 If you want to handle REST Requests, implement the `ActionExtension` interface and override the `getExtensionRestHandlers()` method.  Pass a list of classes which will handle those requests.
 
 ```java
-import org.opensearch.sdk.ActionExtension;
+import org.opensearch.sdk.api.*;
```
Comment (dbwiddis) [inline CREATE_YOUR_FIRST_EXTENSION.md:109]: We prohibit star imports in our code, so just make this `org.opensearch.sdk.api.ActionExtension`.

**Review by dbwiddis — CHANGES_REQUESTED** [review]
One last change please!

**Review by owaiskazi19 — DISMISSED** [review]
Just one change that @dbwiddis has pointed already. Rest LGTM! Thanks @kokibas for cleaning it up

**Review by dbwiddis — APPROVED** [review]
LGTM!

**Review by owaiskazi19 — APPROVED** [review]

**Pull request merged**